bin/BuildPackages.sh

```diff
@@ -79,115 +135,197 @@ build_carat() {
 # It is not possible to move around compiled binaries because these have the
 # path to some data files burned in.
 zcat carat-2.1b1.tgz | tar pxf -
-ln -s carat-2.1b1/bin bin
+# If the symbolic link does not exist then create it.
+# Does not check if bin exists already with some content or as a file, etc.
+# Thus this code is unstable, but should be corrected by package author
+# and not by this script.
+if [[ ! -L ./bin ]]
+then
+  ln -s ./carat-2.1b1/bin ./bin
+fi
 cd carat-2.1b1
-make TOPDIR=`pwd`
+make TOPDIR="$(pwd)"
 chmod -R a+rX .
 cd bin
-aa=`./config.guess`
-for x in "`ls -d1 $GAPDIR/bin/${aa}*`"; do
- ln -s "$aa" "`basename $x`"
-done
+# This sets GAParch as it should be.
+# Alternatively, one could check $GAPDIR/bin for all directories instead.
+source "$GAPDIR/sysinfo.gap"
+# If the symbolic link does not exist then create it.
+if [[ ! -L ./"$GAParch" ]]
+then
+  shopt -s nullglob
+  # We assume that there is only one appropriate directory....
+  for archdir in *
+  do
+    if [[ -d "$archdir" ]] && [[ ! -L "$archdir" ]]
+    then
+      ln -s ./"$archdir" ./"$GAParch"
+    fi
+  done
+  shopt -u nullglob
+fi
```

Comment:
Well, your new code does something different than the old code! And that code tries to allow people with multiple GAP build configs (a little known and little used feature, I believe) on a single computer to build carat once, and use it with all build configs.

On any given GAP installation, the default build config is named "default", and comes in 32 and 64 bit flavors. If `config.guess` returns `FOOBAR`, then GAP puts the compilation results into `bin/FOOBAR-default64` for 64-bit builds, and `bin/FOOBAR-default32` for 32-bit builds. You might in addition have a "debug" build config, then its binaries would end up in `bin/FOOBAR-debug64` resp. `bin/FOOBAR-debug32`. You can have all these binaries simultaneously, which can be handy if you want to switch quickly between different build settings. Also, if your GAP root is on a network share, it might be used by another machine with a different architecture, say `QUUX`, which puts its binaries into `bin/QUUX-default64` etc. 

When a GAP package containing an external executable (such as carat), it uses `DirectoriesPackagePrograms( "carat" )` to get a directory object pointing to the `bin/GARPACH-CONFIGNAME` directory inside the package; this is used to locate and launch the executables.

Now, the above code (or rather, its original version) does the following: Carat puts its binaries into `bin/GAPARCH`. It then proceds to create symlinks`bin/GAPARCH-CONFIGNAME` pointing to `bin/GAPARCH` for all configs it can find.

Your new doesn't do that anymore...